api/layer/multipart_upload.go

```diff
@@ -1184,3 +1250,213 @@ func (n *layer) manualSlice(ctx context.Context, bktInfo *data.BucketInfo, prm P
 
 	return id, elements, nil
 }
+
+// uploadPartAsSlot uploads multipart part, but without correct link to previous part because we don't have it.
+// It uses zero part as pivot. Actual link will be set on CompleteMultipart.
+func (n *layer) uploadPartAsSlot(ctx context.Context, params uploadPartAsSlotParams) (*data.ObjectInfo, error) {
+	zeroPart, err := n.treeService.GetPartByNumber(ctx, params.bktInfo, params.multipartInfo.ID, 0)
+	if err != nil {
+		return nil, fmt.Errorf("get part by number: %w", err)
+	}
+
+	var (
+		id              oid.ID
+		chunk           *[]byte
+		elements        []data.LinkObjectPayload
+		isReturnToPool  bool
+		splitFirstID    = zeroPart.OID
+		splitPreviousID = zeroPart.OID
```

Comment:
Wow. So it's like a tree starting from the first OID? The idea was to make them completely separate. But if this works maybe it's a better scheme in the end (somewhat easier to clean up).

Reply:
It is working for the `Slot` part and helps to avoid double reupload